app/controllers/enrollments_controller.rb

```diff
@@ -6,11 +6,14 @@ class EnrollmentsController < ApplicationController
   before_action :set_mission, only: %i[create destroy]
 
   def create
-    return flash[:alert] = translate('.max_member_count_reached') if member_slots_full?
+    create_transaction = Enrollments::CreateTransaction.new.with_step_args(
+      validate: [mission: @mission],
+      prepare: [mission: @mission]
+    ).call(permitted_params)
 
-    @enrollment = Enrollment.create(enrollment_params)
-    user_feedback_on_creation
+    return flash[:alert] = create_transaction.failure unless create_transaction.success?
```

Comment:
Are you setting the full `failure` object as the error message?
Or is the message properly set on transaction failure?

Reply:
The message is rendered properly on view